dlt/destinations/dataset/dataset.py

```diff
@@ -121,6 +119,49 @@ def _ensure_client_and_schema(self) -> None:
     def __call__(self, query: Any) -> ReadableDBAPIRelation:
         return ReadableDBAPIRelation(readable_dataset=self, provided_query=query)  # type: ignore[abstract]
 
+    def _filter_using_root_table(
```

Comment:
I think the ticket maybe was not completely clear, but we only need these things for the ReadableIbisRelation. It will be quite a bit of work to get the pure sql statements to work on all destinations we support I think, so you can remove all the methods you defined on the non ibis relation, or rather throw an exception here that says Ibis is required to use these.